scripts/download_image.py

```diff
@@ -47,13 +48,25 @@ def __init__(self, total: int, report_function: callable, *args, **kwargs):
                                                  **kwargs)
         self._validate_report_function()
 
-    def report(self, message, multiline=True, *args, **kwargs):
-        if not multiline:
+    def report(self, message: str, next_line: bool = True, *args, **kwargs):
```

Comment:
message: str here but type check fail, reporter functions message: FutureResult

Reply:
namedstuple for annotation type check does not work thought, (do not know field types)
new Python has data type instead of namedtupe. that's work for type check.
personally, I feel do not need update python for so unnecessary new feature.
just note the info.

Reply:
Updated message to future_result. Removed Reporter class and all of it is methods.